lib/accumulate_distribute/util/get_order_log_data.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+const hasOBRequirement = require('./has_ob_requirement')
+const hasIndicatorOffset = require('./has_indicator_offset')
+const hasIndicatorCap = require('./has_indicator_cap')
+const hasTradeRequirement = require('./has_trade_requirement')
+
+const orderMapIndex = {
+  cid: 0,
+  type: 1,
+  orderAmount: 2,
+  orderPrice: 3,
+  dateTime: 4,
+  topAskOrderId: 5,
+  topAskPrice: 6,
+  topBidOrderId: 7,
+  topBidPrice: 8,
+  midPrice: 9,
+  offsetIndicatorValue: 10,
+  capIndicatorValue: 11,
+  lastTradeId: 12,
+  lastTradeMts: 13,
+  lastTradeAmount: 14,
+  lastTradePrice: 15
+}
+
+module.exports = (order, state) => {
+  const { args = {}, lastBook, offsetIndicator, capIndicator, lastTrade } = state
+
+  const data = new Array(16).fill(null)
+  data[orderMapIndex.dateTime] = new Date()
+
+  const { cid, type, amount: orderAmount, price: orderPrice } = order
+
+  data[orderMapIndex.cid] = cid
+  data[orderMapIndex.type] = type
+  data[orderMapIndex.orderAmount] = orderAmount
+  data[orderMapIndex.orderPrice] = orderPrice
+
+  if (hasOBRequirement(args) && lastBook) {
+    const [topAskOrderId = null] = lastBook.topAskLevel() || []
+    const [topBidOrderId = null] = lastBook.topBidLevel() || []
```

Comment:
why `[topBidOrderId = null]` AND `|| []`

Reply:
https://github.com/bitfinexcom/bfx-api-node-models/blob/e10007c7a78a46ad1ccefc806b2cd43ef54c6d64/lib/order_book.js#L319

- destructing for 0 index element for the order id. so, [topAskOrderId = null]
- topAskLevel() might return null or an array. so, `lastBook.topAskLevel() || []`